src/Streamly/Internal/FileSystem/Event/Windows.hs

```diff
@@ -483,7 +497,17 @@ getRelPath Event{..} = eventRelPath
 -- /Internal/
 --
 getRoot :: Event -> String
-getRoot Event{..} = eventRootPath
+getRoot Event{..} = ensureTrailingSlash eventRootPath
+
+-- XXX Change the type to Array Word8 to make it compatible with other APIs.
+--
+-- | Get the absolute file system object path for which the event is generated.
+-- The path is a UTF-8 encoded array of bytes.
+--
+-- /Internal/
+--
+getAbsPath :: Event -> String
+getAbsPath ev = getRoot ev <> getRelPath ev
```

Comment:
How is this compiling?
```
getRelPath ev :: Array Word8
getRoot ev :: Array Word8
(<>) :: Array Word8 -> Array Word8 -> Array Word8
```

Reply:
in Windows return types are String, 
getRelPath :: Event -> String
getRoot :: Event -> String
getAbsPath :: Event -> String